src/clj_cron_parse/core.clj

```diff
@@ -349,15 +367,17 @@
          "
 
   ([now cron]
-   (if-let [{:keys [dom dow] :as cron-map} (make-cron-map cron)]
-     (match [dom dow]
-       [:star :star] (next-date-by-dom now cron-map)
-       [_ :star] (next-date-by-dom now cron-map)
-       [:star _] (next-date-by-dow now cron-map)
-       :else (let [by-dom (next-date-by-dom now cron-map)
-                   by-dow (next-date-by-dow now cron-map)]
-               (-> [by-dom by-dow] sort first)))
-     nil))
+   (when-let [result (when-let [{:keys [dom dow] :as cron-map} (make-cron-map cron)]
+                       (match [dom dow]
+                         [:star :star] (next-date-by-dom now cron-map)
+                         [_ :star] (next-date-by-dom now cron-map)
+                         [:star _] (next-date-by-dow now cron-map)
+                         :else (let [by-dom (next-date-by-dom now cron-map)
+                                     by-dow (next-date-by-dow now cron-map)]
+                                 (-> [by-dom by-dow] sort first))))]
+     (assert (not (t/before? result now))
+             (format "Next cron resulted in a timestamp that was before the provided timestamp, schedule: %s relative to: %s result: %s" cron now result))
```

Comment:
This will cause things to fail _loudly_ but will prevent the issue we saw where we just keep processing a cron job over and over again because the next timestamp is in the past